source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.deployFunctionWithVpc.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import { App, Stack } from "@aws-cdk/core";
+import { LambdaToElasticSearchAndKibana } from "../lib";
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as ec2 from '@aws-cdk/aws-ec2';
+
+// Setup
+const app = new App();
+const stack = new Stack(app, 'test-lambda-elasticsearch-kibana-stack3');
+
+const lambdaProps: lambda.FunctionProps = {
+  code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+  runtime: lambda.Runtime.NODEJS_12_X,
+  handler: 'index.handler',
+};
+
+const vpcProps = {
+  natGateways: 1,
+  natGatewaySubnets: {
+    subnetType: ec2.SubnetType.PUBLIC
+  },
+  enableDnsHostnames: true,
+  enableDnsSupport: true,
+  maxAzs: 1,
+  subnetConfiguration: [
+    {
+      cidrMask: 18,
+      name: "private",
+      subnetType: ec2.SubnetType.PRIVATE,
+    },
+    {
+      cidrMask: 18,
+      name: "public",
+      subnetType: ec2.SubnetType.PUBLIC,
+    }
+  ]
+};
+
+new LambdaToElasticSearchAndKibana(stack, 'test-lambda-elasticsearch-kibana3', {
+  lambdaFunctionProps: lambdaProps,
+  domainName: "deploywithvpctest",
+  deployVpc: true,
+  vpcProps
```

Comment:
How about a test that provides deployVpc and no props (an accepted scenario). The construct should create a default VPC that is appropriate for its use.

Reply:
Currently in index.ts, the vpc is only added on lambda(ElasticSearch cluster remain staying in public), so to successfully call ElasticSearch cluster, the vpc needed to be configured (using vpcProps) to make the call from lambda. If user provides deployVpc with no props and still want to make a successful call, we may need to override esDomainProps to add vpc to ElasticSearch cluster as well. Is that the next step I should go with?